include/opencv2/opencv.hpp

```diff
@@ -102,4 +101,39 @@
 #include "opencv2/viz.hpp"
 #endif
 
+// Finally CUDA specific entries are checked and added
+#ifdef HAVE_OPENCV_CUDAARITHM
+#include "opencv2/cudaarithm.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDABGSEGM
+#include "opencv2/cudabgsegm.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDACODEC
+#include "opencv2/cudacodec.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDAFEATURES2D
+#include "opencv2/cudafeatures2d.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDAFILTERS
+#include "opencv2/cudafilters.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDAIMGPROC
+#include "opencv2/cudaimgproc.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDALEGACY
+#include "opencv2/cudalegacy.hpp"
```

Comment:
Agreed that it would be better not to standard include cuda legacy code. That being said, if people wanted to use other CUDA modules, shouldn't they always provide the include path for CUDA to the compiler? To me that seems common sense!